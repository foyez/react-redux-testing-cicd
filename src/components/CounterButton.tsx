// import React, { useState } from 'react'

// export const CounterButton = () => {
//   const [count, setCount] = useState(0)

//   // return <button onClick={() => setCount(count + 1)}>Count: {count}</button>
//   return (
//     <Button id="counter" onClick={() => setCount(count + 1)} count={count} />
//   )
// }

// interface IButton {
//   id: string
//   count: number
//   onClick: () => void
// }

// const Button: React.FC<IButton> = React.memo(
//   ({ id, count, onClick }) => (
//     <button id={id} onClick={onClick}>
//       Count: {count}
//     </button>
//   ),
//   (prevProps, nextProps) => {
//     console.log(prevProps.count === nextProps.count)
//     return prevProps.count === nextProps.count
//   },
// )

import React from 'react'

type CounterButtonProps = {
  color: string
}

type CounterButtonState = {
  count: number
}

export class CounterButton extends React.Component<
  CounterButtonProps,
  CounterButtonState
> {
  state = {
    count: 0,
  }

  shouldComponentUpdate(nextProps: any, nextState: any) {
    return this.state.count !== nextState.count
  }

  render() {
    return (
      <button
        id="counter"
        color={this.props.color}
        onClick={() => this.setState((state) => ({ count: state.count + 1 }))}
      >
        Count: {this.state.count}
      </button>
    )
  }
}
