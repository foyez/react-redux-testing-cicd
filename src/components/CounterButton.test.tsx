import React from 'react'
import { shallow } from 'enzyme'

import { CounterButton } from './CounterButton'

describe('CounterButton Component', () => {
  const color = 'red'
  const wrapper = shallow(<CounterButton color={color} />)

  it('expects to render CounterButton component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('correctly increments the counter', () => {
    wrapper.find('#counter').simulate('click')
    wrapper.find('#counter').simulate('click')
    expect(wrapper.state()).toEqual({ count: 2 })

    wrapper.find('#counter').simulate('click')
    expect(wrapper.state()).toEqual({ count: 3 })

    expect(wrapper.props().color).toEqual('red')
  })
})
