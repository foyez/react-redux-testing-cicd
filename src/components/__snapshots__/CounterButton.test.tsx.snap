// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton Component expects to render CounterButton component 1`] = `ShallowWrapper {}`;
